include/highfive/bits/H5Easy_misc.hpp

```diff
@@ -0,0 +1,97 @@
+/*
+ *  Copyright (c), 2017, Adrien Devresse <adrien.devresse@epfl.ch>
+ *
+ *  Distributed under the Boost Software License, Version 1.0.
+ *    (See accompanying file LICENSE_1_0.txt or copy at
+ *          http://www.boost.org/LICENSE_1_0.txt)
+ *
+ */
+#ifndef H5EASY_BITS_MISC_HPP
+#define H5EASY_BITS_MISC_HPP
+
+#include "../H5Easy.hpp"
+
+namespace HighFive
+{
+    namespace detail
+    {
+        inline auto error(const HighFive::File& file, const std::string& path, std::string message)
+        {
+            message += "\n";
+            message += "path: '" + path + "'\n";
+            message += "filename: '" + file.getName() + "'\n";
+
+            return std::runtime_error(message);
+        }
+    }
+
+    inline std::string groupName(const std::string& path)
+    {
+        std::size_t idx = path.find_last_of("/\\");
+
+        if ( idx == std::string::npos )
+        {
+            return "/";
+        }
+
+        if ( idx == 0 )
+        {
+            return "/";
+        }
+
+        return path.substr(0,idx);
+    }
+
+    inline void createGroupsToDataSet(HighFive::File& file, const std::string& path)
+    {
+        std::string group_name = groupName(path);
+
+        if ( not file.exist(group_name) )
+        {
+            file.createGroup(group_name);
+        }
+    }
+
+    inline auto size(const HighFive::File& file, const std::string& path)
+    {
+        HighFive::DataSet dataset = file.getDataSet(path);
+
+        auto dataspace = dataset.getSpace();
+
+        auto dims = dataspace.getDimensions();
+
+        using T = decltype(dims)::value_type;
+
+        return std::accumulate(dims.begin(), dims.end(), 1, std::multiplies<T>());
+    }
+
+    inline auto size(const HighFive::DataSet& dataset)
```

Comment:
I am not convinced on having these helper functions. To me it seems like going away from an OO paradigm. They very useful shortcuts though, I would implement them in the core H5DataSet (and probably internally cache the results). What do you think? @tristan0x

Reply:
I think the load, dump, shape, and size are simple operations that deserve to be in this simple API.
On the other hand, I am not completely convinced for `groupName` and `createGroupsToDataSet`:
* `groupName` is a very basic function with no H5 specificities.
* `createGroupsToDataSet` is not meant to be used in simple use-cases like the load/dump functions for instance. So this operation should be done using the OO API.

Reply:
If `createGroupsToDataSet` is internal, then it should not be usable.

> Can you specify what you mean by "So this operation should be done using the OO API."
I meant that this helper is not necessary and if the user needs to do it, it should use what current API already provides.

Reply:
I see. Is there a function that does exactly this in the API? I.e. recursively create all groups needed to create a DataSet. E.g. recursively create groups `/path/to/a/certain` if the input path to the DataSet is `/path/to/a/certain/dataset`?

Reply:
@ferdonline The problem is that `createGroup` throws if called for an existing group. Otherwise I would do what you suggest.